### PR TITLE
fix: add aliases to /curio/overview page, to prevent broken links

### PR DESCRIPTION
**Pull request by rjan90**
Start on #757

Just added the reported broken link, there are probably a ton of broken links due to the Move Curio Docs PR: https://github.com/filecoin-project/lotus-docs/pull/744.

@LexLuthr & @Reiers feel free to take over this and restore any curio-links to link to the overview page instead.

### CODE REVIEW

**Comment by LexLuthr** [comment]
We should fix the original links to point to Curio docs instead of creating a link chain via Lotus docs.
https://github.com/filecoin-project/lotus-docs/issues/757#issuecomment-2315057381

**Comment by smagdali** [comment]
Someone has access to 404 logs to check? 

Also the live page is now giving me a 500 instead of a 404 and I don't understand cos this isn't merged yet!? (also looks fine to me) 

image

File: content/en/storage-providers/curio/overview.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ menu:
         parent: "curio"
 aliases:
   - /storage-providers/lotus-provider/overview
+  - /storage-providers/lotus-provider/setup
```
Comment (LexLuthr) [inline content/en/storage-providers/curio/overview.md:10]:
```suggestion
  - /storage-providers/lotus-provider/setup
  - /storage-providers/curio/cli
  - /storage-providers/curio/design
  - /storage-providers/curio/setup
  - /storage-providers/curio/config
  - /storage-providers/curio/install
  - /storage-providers/curio/local-network
  - /storage-providers/curio/sealing
  - /storage-providers/curio/tasks
  - /storage-providers/curio/ui
```

File: content/en/storage-providers/curio/_index.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - /storage-providers/curio/install
+  - /storage-providers/curio/local-network
+  - /storage-providers/curio/sealing
+  - /storage-providers/curio/tasks
+  - /storage-providers/curio/ui
+weight: 105
+toc: true
+---
+
+<p>If you are not redirected, <a href="https://docs.curiostorage.org">click here</a>.</p>
```
Comment (LexLuthr) [inline content/en/storage-providers/curio/_index.md:24]:
```suggestion
<p>Curio documents have been moved. If you are not redirected, <a href="https://docs.curiostorage.org">click here</a>.</p>
```

**Review by LexLuthr — APPROVED** [review]

**Pull request merged**